src/drivers/vga/text_mode.rs

```diff
@@ -91,7 +94,16 @@ impl Writer {
                 // Actually write the character to the screen, escapes have been handled
                 // previously no need to worry about those anymore.
                 let pos: u16 = (self.y * WIDTH + self.x) as u16;
-                let byte: u16 = ((char.style as u16) << 8) | u16::from(char.ascii);
+
+                // Checks whether ch is a non-ascii character
+                // If it's not ascii, sets it to a space (0x20)
+                // If it's ascii nothing happens
+                let byte: u16 = if ch > ASCII_MAX || ch < ASCII_MIN {
+                    u16::from(b' ')
```

Comment:
needs to have the style because the background color of the char is configurable